policy-controller/k8s/index/src/index.rs

```diff
@@ -1146,10 +1195,20 @@ impl PolicyIndex {
         name: String,
         server: &server::Server,
         authentications: &AuthenticationNsIndex,
+        probes: &PortMap<HashSet<String>>,
+        port: NonZeroU16,
     ) -> InboundServer {
         tracing::trace!(%name, ?server, "Creating inbound server");
         let authorizations = self.client_authzs(&name, server, authentications);
-        let routes = self.http_routes(&name, authentications);
+        let mut routes = self.http_routes(&name, authentications);
```

Comment:
TIOLI: I bias to avoiding mut where I can:
```
let routes = {
  let r = self.http_routes(&name, authentications);
  match self.cluster_info.probe_networks {
    Some(ref probe_networks) if r.is_empty() => Pod::http_probe_routes(...),
    _ => r,
  }
};  
```

Reply:
https://github.com/linkerd/linkerd2/blob/a93c01e10a83043b8c01897827b30badec70ea9a/policy-controller/k8s/index/src/index.rs#L1205-L1210